x/emissions/keeper/msgserver/msg_server_worker_payload.go

```diff
@@ -22,25 +20,15 @@ func (ms msgServer) VerifyAndInsertInferencesFromTopInferers(
 	workerDataBundles []*types.WorkerDataBundle,
 	maxTopWorkersToReward uint64,
 ) (map[string]bool, error) {
-
 	inferencesByInferer := make(map[string]*types.Inference)
 	latestInfererScores := make(map[string]types.Score)
 	for _, workerDataBundle := range workerDataBundles {
 		/// Do filters first, then consider the inferenes for inclusion
 		/// Do filters on the per payload first, then on each inferer
 		/// All filters should be done in order of increasing computational complexity
 
-		// check signatures from the bundle throw if invalid!
-		pk, err := hex.DecodeString(workerDataBundle.Pubkey)
-		if err != nil || len(pk) != secp256k1.PubKeySize {
-			return nil, types.ErrSignatureVerificationFailed
-		}
-		pubkey := secp256k1.PubKey(pk)
-
-		src := make([]byte, 0)
-		src, _ = workerDataBundle.InferenceForecastsBundle.XXX_Marshal(src, true)
-		if !pubkey.VerifySignature(src, workerDataBundle.InferencesForecastsBundleSignature) {
-			return nil, types.ErrSignatureVerificationFailed
+		if err := workerDataBundle.Validate(); err != nil {
+			continue // Ignore only invalid worker data bundles
```

Comment:
but is it right to silently ignore, or should we be ignoring, continuing, and then raising error?

what do we do if we raise the error, what will the caller do with said error? log and move on?

Reply:
I'm team "silent ignore" because the leader will be committing txs anyway, so already the logs are likely to be 1 anon away from the worker who produced the failing bundle. And don't think we have ability to emit events like in the EVM => relying on an explorer wouldn't help the worker gain insight either.